### ISSUE TITLE
[REQ] Data and Structure Syncronization

### ISSUE THREAD

**Issue by ghost** [sequelpro/sequelpro#1155]
Issue imported from Google Code: https://code.google.com/p/sequel-pro/issues/detail?id=1155
Reporter:highlander.44@gmail.com Date:2011-08-19 08:31:54 Status:Duplicate Closed:2011-08-01

For reference, check out the SQLYog tool for windows -- <a href="http://www.webyog.com/en/index.php" rel="nofollow">http://www.webyog.com/en/index.php</a>

Game Changing functionality that we need on the Mac side:
1. Compare 2 Databases for structural differences (Schema Synchronization)
   This compares all of the tables from one DB to another and reports out the SQL Statements to 
    get them identical (create, alter table, create index, etc).   This allows you make changes, say
    to your &quot;development&quot; database and once you're happy, roll all of those changes out to 
    your production database by having the tool do the comparisons and give you the SQL to run
2. Data Synchronization
   Basically the same concept:  Compare the data in every table between two databases.  Output
   all of the INSERT/UPDATE/DELETE statements to make them the same.

In either scenario, you select one dB to be the master and compare the 2nd database to that one, so that after a sync, the 2nd DB would look like the first.

We use SQLYog to do comparisons both ways so that we can easily create both Deploy and Rollback
SQL Scripts.

I wish that Sequel Pro on the Mac did this.  As it currently is, I need to open up VMWare and run a 
Windows Instance so that I can run SQLYog and then copy those sync scripts back and forth.  Kind of 
a pain.  Sequel Pro is already the best tool out there, this would take it to an entirely new level.

**Comment by geoidesic** [comment]
Bump! This missing feature is the only reason I still have to install Wine (which works better for sqlyog than a virtual machine btw).

**Comment by phenixiim** [comment]
Yeees please, at least the structure, pleeeeese :D
